libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -69,35 +71,47 @@ instance ToByteString TeamFeatureName where
 typeTeamFeatureName :: Doc.DataType
 typeTeamFeatureName = Doc.string . Doc.enum $ cs . toByteString' <$> [(minBound :: TeamFeatureName) ..]
 
-data TeamFeatureStatus = TeamFeatureEnabled | TeamFeatureDisabled
-  deriving stock (Eq, Show, Generic)
-  deriving (Arbitrary) via (GenericUniform TeamFeatureStatus)
+newtype TeamFeatureStatus = TeamFeatureStatus
+  {teamFeatureStatusValue :: TeamFeatureStatusValue}
```

Comment:
*sigh*

This is rolling back the change we did in tag `image/2.83.22` (wireapp/wire-server#1122).  The reason we changed it was that it made the code simpler all over the backend, as you can tell from all the awkwardness you had to add.  :-P

So, good thing you asked for a second review!

Reply:
we discussed this, and came to the conclusion that this code is more beaurocratic, but less surprising.  so we'll keep it.